OneSignalSDK/unittest/src/test/java/com/test/onesignal/NotificationOpenedActivityHMSIntegrationTestsRunner.java

```diff
@@ -0,0 +1,125 @@
+package com.test.onesignal;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.support.annotation.NonNull;
+
+import com.onesignal.NotificationOpenedActivityHMS;
+import com.onesignal.OneSignal;
+import com.onesignal.ShadowCustomTabsClient;
+import com.onesignal.ShadowCustomTabsSession;
+import com.onesignal.ShadowOSUtils;
+import com.onesignal.ShadowOSViewUtils;
+import com.onesignal.ShadowOSWebView;
+import com.onesignal.ShadowOneSignalRestClient;
+import com.onesignal.StaticResetHelper;
+import com.onesignal.example.BlankActivity;
+
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.Robolectric;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.annotation.Config;
+import org.robolectric.shadows.ShadowLog;
+
+import java.util.UUID;
+
+import static com.onesignal.InAppMessagingHelpers.ONESIGNAL_APP_ID;
+import static com.test.onesignal.TestHelpers.fastColdRestartApp;
+import static com.test.onesignal.TestHelpers.threadAndTaskWait;
+import static junit.framework.Assert.assertEquals;
+import static org.robolectric.Shadows.shadowOf;
+
+
+@Config(packageName = "com.onesignal.example",
+    instrumentedPackages = { "com.onesignal" },
+    shadows = {
+        ShadowOSUtils.class,
+        ShadowOneSignalRestClient.class,
+        ShadowCustomTabsClient.class,
+        ShadowOSWebView.class,
+        ShadowOSViewUtils.class,
+        ShadowCustomTabsClient.class,
+        ShadowCustomTabsSession.class
+    },
+    sdk = 26
+)
+
+
+@RunWith(RobolectricTestRunner.class)
+public class NotificationOpenedActivityHMSIntegrationTestsRunner {
+
+    @BeforeClass // Runs only once, before any tests
+    public static void setUpClass() throws Exception {
+        ShadowLog.stream = System.out;
+        TestHelpers.beforeTestSuite();
+        StaticResetHelper.saveStaticValues();
+    }
+
+    @Before
+    public void beforeEachTest() throws Exception {
+        TestHelpers.beforeTestInitAndCleanup();
+    }
+
+    private static Intent helper_baseHMSOpenIntent() {
+        return new Intent()
+                .setFlags(Intent.FLAG_ACTIVITY_NO_HISTORY | Intent.FLAG_ACTIVITY_CLEAR_WHEN_TASK_RESET | Intent.FLAG_ACTIVITY_MULTIPLE_TASK | Intent.FLAG_ACTIVITY_NEW_TASK)

```

Comment:
this combination Intent.FLAG_ACTIVITY_MULTIPLE_TASK | Intent.FLAG_ACTIVITY_NEW_TASK is forcing to always create a new task, can this lead to having 2 instances at the same time?

Reply:
idem

Reply:
In testing found out that `FLAG_ACTIVITY_NEW_TASK | FLAG_ACTIVITY_MULTIPLE_TASK` provides the requirement of "Allows the app developer to omit foregrounding the app if they choose".

We always want to create a new task so it doesn't resume some existing task. If it resumes an existing task then it won't be possible to prevent the app from showing. Some developer want to process the click in the background without bring the app to the foreground.